tests/PHPStan/Analyser/data/filter-var.php

```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types=1);
+
+namespace FilterVar;
+
+use function PHPStan\Testing\assertType;
+
+class FilterVar
+{
+
+	public function doFoo($mixed): void
+	{
+		assertType('int|false', filter_var($mixed, FILTER_VALIDATE_INT));
+		assertType('int|null', filter_var($mixed, FILTER_VALIDATE_INT, ['flags' => FILTER_NULL_ON_FAILURE]));
+		assertType('array<int|false>', filter_var($mixed, FILTER_VALIDATE_INT, ['flags' => FILTER_FORCE_ARRAY]));
+		assertType('array<int|null>', filter_var($mixed, FILTER_VALIDATE_INT, ['flags' => FILTER_FORCE_ARRAY|FILTER_NULL_ON_FAILURE]));
+		assertType('0|int<17, 19>', filter_var($mixed, FILTER_VALIDATE_INT, ['options' => ['default' => 0, 'min_range' => 17, 'max_range' => 19]]));
+	}
+
+	public function intToInt(int $int, array $options): void
+	{
+		assertType('int', filter_var($int, FILTER_VALIDATE_INT));
+		assertType('int|false', filter_var($int, FILTER_VALIDATE_INT, $options));
+		assertType('int<0, max>|false', filter_var($int, FILTER_VALIDATE_INT, ['options' => ['min_range' => 0]]));
+	}
+
+	public function constants(): void
+	{
+		assertType('array<false>', filter_var(false, FILTER_VALIDATE_BOOLEAN, FILTER_FORCE_ARRAY | FILTER_NULL_ON_FAILURE));
+
+		assertType('17', filter_var(17, FILTER_VALIDATE_INT));
+
+		assertType('17.0', filter_var(17, FILTER_VALIDATE_FLOAT));
+
+		assertType("'17'", filter_var(17));
+		assertType("'17'", filter_var(17.0));
+		assertType("'1'", filter_var(true));
+		assertType("''", filter_var(false));
```

Comment:
Does this also mean?

```suggestion
		assertType("'1'", filter_var(true));
		assertType("''", filter_var(false));
     assertType("''|'1'", filter_var($bool));

```

Reply:
Yes, it does, this is correctly implemented in BoolType already. Will add more tests tomorrow, thx